.github/workflows/build-images-docs-builder.yaml

```diff
@@ -0,0 +1,147 @@
+
+      - name: Set up git
+        run: |
+          git config user.name "Cilium Imagebot"
+          git config user.email "noreply@cilium.io"
+
+      - name: Generate image tag for docs-builder
+        id: docs-builder-tag
+        run: |
+          echo tag="$(git ls-tree --full-tree HEAD -- ./Documentation | awk '{ print $3 }')" >> $GITHUB_OUTPUT
+
+      - name: Check if tag for docs-builder already exists
+        id: docs-builder-tag-in-repositories
+        shell: bash
+        run: |
+          if docker buildx imagetools inspect quay.io/${{ env.QUAY_ORGANIZATION_DEV }}/docs-builder:${{ steps.docs-builder-tag.outputs.tag }} &>/dev/null; then
+            echo exists="true" >> $GITHUB_OUTPUT
+          else
+            echo exists="false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Login to quay.io
+        if: ${{ steps.docs-builder-tag-in-repositories.outputs.exists == 'false' }}
+        uses: docker/login-action@465a07811f14bebb1938fbed4728c6a1ff8901fc # v2.2.0
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_DOCS_BUILDER_USERNAME }}
+          password: ${{ secrets.QUAY_DOCS_BUILDER_PASSWORD }}
+
+      - name: Build docs-builder image
+        if: ${{ steps.docs-builder-tag-in-repositories.outputs.exists == 'false' }}
+        uses: docker/build-push-action@2eb1c1961a95fc15694676618e422e8ba1d63825 # v4.1.1
+        id: docker_build_docs_builder
+        with:
+          provenance: false
+          context: ./Documentation
+          file: ./Documentation/Dockerfile
+          push: true
+          tags: |
+            quay.io/${{ env.QUAY_ORGANIZATION_DEV }}/docs-builder:${{ steps.docs-builder-tag.outputs.tag }}
+
+      - name: Retrieve image digest
+        if: ${{ steps.docs-builder-tag-in-repositories.outputs.exists == 'false' }}
+        shell: bash
+        run: |
+          mkdir -p image-digest/
+          echo "## docs-builder" > image-digest/docs-builder.txt
+          echo "" >> image-digest/docs-builder.txt
+          echo "\`quay.io/${{ env.QUAY_ORGANIZATION_DEV }}/docs-builder:${{ steps.docs-builder-tag.outputs.tag }}@${{ steps.docker_build_docs_builder.outputs.digest }}\`" >> image-digest/docs-builder.txt
+          echo "" >> image-digest/docs-builder.txt
+
+      - name: Upload artifact digests
+        if: ${{ steps.docs-builder-tag-in-repositories.outputs.exists == 'false' }}
+        uses: actions/upload-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce # v3.1.2
+        with:
+          name: image-digest docs-builder
+          path: image-digest
+          retention-days: 1
+
+      - name: Update docs-builder image reference in CI workflow
+        if: ${{ steps.docs-builder-tag-in-repositories.outputs.exists == 'false' }}
+        run: |
+          Documentation/update-docs-builder-image.sh "quay.io/${{ env.QUAY_ORGANIZATION_DEV }}/docs-builder:${{ steps.docs-builder-tag.outputs.tag }}@${{ steps.docker_build_docs_builder.outputs.digest }}"
```

Comment:
Wouldn't this run the not trusted `update-docs-builder-image.sh` from the pull request branch, rather than the default trusted branch? If so, a user could change this script and run whatever they wanted in the workflow.

Reply:
Note that triggering the overall workflow is only done with @cilium/docs-structure approval, so we are partially relying here on those reviewers doing due diligence on any PRs requesting to change build infrastructure.

It would probably be better practice to do some combination of the following:

- Log out of Quay after images are pushed
- Directly call docker from the workflow to wrap & run the docs script

Reply:
Both proposals look good, but I'm not sure how to implement them.

> * Log out of Quay after images are pushed

I don't know how to explicitly logout from Quay. The Action we use supports a `logout` parameter, already enabled by default, but logging out occurs only at the end of the job. One solution I see would be to split the current job into several ones, starting a new one after `Build docs-builder image` (which I think is the only step that requires the log in)?

> * Directly call docker from the workflow to wrap & run the docs script

Do you mean replacing the script invocation with the following?"

```console
$ docker run --rm -v $PWD:/cilium -w /cilium cilium/docs-builder \
    bash -c "git config --global --add safe.directory /cilium && /cilium/Documentation/update-docs-builder-image.sh 'quay.io/...'"
```

Reply:
> One solution I see would be to split the current job into several ones, starting a new one after Build docs-builder image (which I think is the only step that requires the log in)?

I would also be in favor of this approach.